store/sqlstore/channel_store.go

```diff
@@ -1812,6 +1812,69 @@ func (s SqlChannelStore) UpdateLastViewedAt(channelIds []string, userId string)
 	return times, nil
 }
 
+// CountUnreadSince gives the number of posts that a channel has since some date, this doesn't follow any threads.
+// this allows us to know how many posts have bean read already.
+func (s SqlChannelStore) CountUnreadSince(channelID string, since int64) (int64, *model.AppError) {
```

Comment:
Maybe a better name for this would be `CountPostsBefore`? It's kind of doing the opposite of what the name would suggest.

Reply:
Actually, it looks like you might just have the logic in the SQL statement backwards since it's finding posts made before the time, not after

Reply:
good catch! this function changed as I was deciding going one way or the other and ended in a weird place.